test/machineslib.py

```diff
@@ -42,21 +42,16 @@ def waitPageInit(self):
         with self.browser.wait_timeout(30):
             self.browser.wait_in_text("body", "Virtual machines")
 
-    def performAction(self, vmName, action, checkExpectedState=True, connectionName="system"):
+    def performAction(self, vmName, action, checkExpectedState=True, connectionName="system", logPath=None):
         b = self.browser
         m = self.machine
 
-        def getStartTime():
-            return m.execute(f"grep 'starting up' /var/log/libvirt/qemu/{vmName}.log | tail -1")
-
-        initialTime = getStartTime()
         b.click(f"#vm-{vmName}-{connectionName}-action-kebab button")
         b.wait_visible(f"#vm-{vmName}-{connectionName}-action-kebab > .pf-v5-c-dropdown__menu")
         b.click(f"#vm-{vmName}-{connectionName}-{action} a")
-        if (action == "reboot" or "forceReboot") and connectionName == "system":
-            # Check VM doesn't get rebooted after opening dialog
-            # https://bugzilla.redhat.com/show_bug.cgi?id=2221144
-            self.assertEqual(initialTime, getStartTime())
+        if (action == "reboot" or "forceReboot") and connectionName == "system" and logPath:
+            # Rebooting clears the log, wait for that to happen
+            testlib.wait(lambda: "Linux version" not in m.execute(f"cat {logPath}"))
```

Comment:
The `not` makes no sense to me here -- this is already satisfied, as the callers already do the `echo '' > logfile` thing. The only thing that would change here is that "Linux version" appears again, to make sure the boot happened.

Reply:
Ah, you wanted to ensure that boot did *not* happen. So:
 - This doesn't work, it's immediately true due to the synchronous truncation of the file. Please change to a sleep(5) and a single-shot test.
 - Please do that positive test after clicking the modal confirmation, to ensure that the reboot actually works.